demo/src/assets/gltf/assets.rs

```diff
@@ -138,8 +138,8 @@ pub struct MeshAssetData {
 pub struct MeshAssetPart {
     pub opaque_pass: MaterialPass,
     pub opaque_material_descriptor_set: DescriptorSetArc,
-    // These are optional because we might want to disable casting shadows
-    pub shadow_map_pass: Option<MaterialPass>,
+    pub casts_shadows: bool,
+    pub receives_shadows: bool,
```

Comment:
These `casts_shadows` or `receives_shadows` can probably just be removed if the plan is to move them to some higher level `RenderProxy` concept.

Reply:
Do you mean as fields in `MeshRenderNode` or something else?

Reply:
Yeah, something populated by the extract job. I can see a reason why you'd import a model into the world and then, on a per "entity" basis, decide whether or not it should cast shadows with some component. During the extract, at the same time we pull out the model info, we'd also check that component to see if we should cast or receive shadows.

Reply:
Yeah that sounds good!